src/command/mtr-command.ts

```diff
@@ -100,6 +100,11 @@ export class MtrCommand implements CommandInterface<MtrOptions> {
 
 			await cmd;
 			const [hops, rawOutput] = await this.parseResult(result.hops, result.data, true);
+			const lastHop = [...hops].reverse().find(h => h.resolvedAddress && !h.duplicate);
```

Comment:
That's not exactly how it works with MTR. We've battled with that issue extensively with @jimaek.

MTR repeats pings to the last record multiple times, and reports them as new entries. It's no as noticible in standard mode, because they filter it out, it becomes a problem when you attempt to output the query in a non-standard way. `--raw` in this case.

Reply:
https://bugs.launchpad.net/mtr/+bug/928509

Reply:
I suppose if that hop times out it would happen? That's exactly a case where we should NOT return a previous IP because it won't match the user's input.

Reply:
The problem is, it times out on duplicate queries while we already have the response. It’s especially noticeable on queries to distant DNS, it will return the actual target, but then will continue pinging and timing out, resulting in a litany of empty records.

Reply:
In that case wouldn't `.find(h => !h.duplicate)` be sufficient?

Reply:
No. Because ‘duplicate: true’ is defined when the ‘resolvedAddress’ comes back, and such an address is already defined. Anything else - we assume will come at some point and hence we can’t flag as duplicates.